packages/design-system/src/components/Table/TableCell.jsx

```diff
@@ -0,0 +1,84 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import classNames from 'classnames';
+
+export const TableCell = ({
+  children,
+  className,
+  data,
+  stackedTitle,
+  stackedClassName,
+  type,
+  ...attributeOptions
+}) => {
+  const classes = classNames(
+    'ds-c-table__cell',
+    className,
+    type === 'numeric' ? 'ds-c-table__cell--numeric' : null
+  );
+
+  const stackedClasses = classNames(
+    'ds-c-table--stacked__col-header',
+    'ds-u-font-weight--bold',
+    stackedClassName
+  );
+
+  const renderStackedTitle = () => {
+    const isValidStackedTitle = stackedTitle && stackedTitle.length > 0;
+
+    return (
+      { isValidStackedTitle } && (
+        <span aria-hidden="true" className={stackedClasses}>
+          {stackedTitle}
+        </span>
+      )
+    );
+  };
+
+  return (
+    <td className={classes} role="cell" {...attributeOptions}>
+      {renderStackedTitle()}
+      {data}
+      {children}
+    </td>
+  );
+};
+
+TableCell.defaultProps = {
+  className: '',
+  stackedClassName: '',
+  type: 'text',
+};
+
+TableCell.propTypes = {
+  /**
+   * The table cell contents.
+   */
+  children: PropTypes.node,
+  /**
+   * Additional classes to be added to the table cell element.
+   */
+  className: PropTypes.string,
+  /**
+   * The table cell data.
+   */
+  data: PropTypes.node.isRequired,
```

Comment:
What's the difference between `data` and `children`?

Reply:
Good point, i have not used the children for the cell, instead have put the cell value in data. Will explore using children only and maybe keep data for customised rendering?

Reply:
I think it makes sense to only use `children`, as any value inside `children` will just render inside `TableCell` which I think is what we want. What do you mean by customized rendering?

Reply:
I was thinking adding an option via prop to pass a render function for customizing the children data. But decided not to as the children value should contain renderable data already.

Reply:
Yeah, I think `data` was left over from when we would pass in objects containing table data. Now that we are doing the approach of `children`, i dont think its necessary